### PR TITLE
Story/4017 rework3

### PR DESCRIPTION
**Pull request by joe-gb**
Adding an activity status to edi.document to contain the
current activity (preparing, executing) of the document.

EDI gateway transfers commit explicitly so that each
stage of transfer>prepare>execute are visible, a
rollback will only occur for a subsection of this
transaction - a failed execution will now leave a
prepared document that can be viewed in the UI.

### CODE REVIEW

File: addons/edi/models/edi_document.py [hunk 1]
```diff
@@ -303,12 +312,40 @@ def execute_records(self):
...
                              (stats.count / count))
 
+    @contextlib.contextmanager
+    @api.multi
+    def manage_activity(self, activity):
+        """Sets the current activity of the document so that a
+        user can see the activity while the enclosed action is running."""
+        self.ensure_one()
+        try:
+            # Write activity status and commit it
```
Comment (spenced) [inline addons/edi/models/edi_document.py:322]: Comment doesn't add anything!

File: addons/edi/views/edi_document_views.xml [hunk 2]
```diff
@@ -164,6 +164,13 @@
 		  </div>
 		  <div class="col-xs-6">
 		    <span class="pull-right text-right">
+		      <field name="activity"
+		       widget="label_selection"
+		       options="{'classes': {
+		      		'cancel': 'danger',
```
Comment (spenced) [inline addons/edi/views/edi_document_views.xml:170]: Why 'danger'?

Comment (joe-gb) [reply addons/edi/views/edi_document_views.xml]: "danger" any information in front of you may change as something is currently happening to this document, do not touch, hands off, you may end the world? :)

File: addons/edi/models/edi_document.py [hunk 3]
```diff
@@ -370,12 +412,17 @@ def action_execute(self):
...
+
+    @api.multi
+    def _action_execute(self):
         self.ensure_one()
         # Lock document
         self.lock_for_action()
         # Automatically prepare document if needed
         if self.state == 'draft':
-            prepared = self.action_prepare()
+            prepared = self._action_prepare()
```
Comment (mcb30) [inline addons/edi/models/edi_document.py:425]: Does this mean that the UI will show just "executing" for both preparation and execution steps?

File: addons/edi/models/edi_transfer.py [hunk 4]
```diff
@@ -177,6 +180,9 @@ def do_transfer(self, conn):
                              self.gateway_id.name, doc.name)
                 prepared = doc.action_prepare()
                 if prepared:
+                    # Document is prepared, we can commit this before beginning
+                    # execution.
+                    self.env.cr.commit()
```
Comment (mcb30) [inline addons/edi/models/edi_transfer.py:185]: This seems to be redundant, since action_prepare() will commit anyway.

**Pull request closed without merging**